scripts/src/internal/services.ts

```diff
@@ -178,8 +189,14 @@ export async function paymentComplete(payment: CompletedPayment) {
 	const prevStatus = order.status;
 	const prevStatusDate = order.currentStatusDate;
 	order.setStatus("completed");
+
 	await order.save();
 
+	// if it was an incoming transfer, clear the cache so we wont process again
+	if (incomingOrderId) {
+		await redis.async.del(payment.id);
```

Comment:
I think you might be able to skip this if you use `setex` to set this key.

Reply:
I can, however it still won't invalidate the cache once the hook triggers, and won't protect from a 2nd trigger, so even if I use setex I think we should manually invalidate, don't you think?